### PR TITLE
PAYARA-1394 OAuth 2 client integration

### PR DESCRIPTION
**Pull request by Cousjava**

### CODE REVIEW

**Comment by Cousjava** [comment]
Jenkins test please

**Comment by Cousjava** [comment]
Jenkins test please

File: api/payara-api/src/main/java/fish/payara/security/oauth2/api/OAuthIdentityStore.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    
+    
+    
+    /**
+     * This method will always return a valid result
+     * @param credentials
+     * @return 
+     */
+    @Override
+    public CredentialValidationResult validate(Credential credentials){
```
Comment (arjantijms) [inline api/payara-api/src/main/java/fish/payara/security/oauth2/api/OAuthIdentityStore.java]: This method shouldn't be needed as the interface already contains a default method for this (indeed, examples still don't always use that interface method)

**Comment by Cousjava** [comment]
Jenkins test please

**Comment by Cousjava** [comment]
Jenkins test please

**Comment by Cousjava** [comment]
Jenkins test please

File: api/payara-api/src/main/java/fish/payara/security/oauth2/annotation/OAuth2AuthenticationDefinition.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+public @interface OAuth2AuthenticationDefinition {
+    
+    /**
+     * Required. The URL for the OAuth2 provider to provide authentication
+     * <p>
+     * This must be a https endpoint
+     * @return 
+     */
+    @NotNull
+    String authEndpoint();
```
Comment (MarkWareham) [inline api/payara-api/src/main/java/fish/payara/security/oauth2/annotation/OAuth2AuthenticationDefinition.java]: Please put a full name in here, not 'auth'

**Pull request merged**